### PR TITLE
Fix `embedding_wrapper.c` for Julia 1.11; also, run CI on 1.11, but skip the `incremental=false` tests on 1.11

### LINKED ISSUE #951: Incompatibility with Julia 1.11-rc1 and 1.11-rc2
Creating an app using Julia 1.11-rc1 (also the manifest files pointing to 1.11-rc1) fails with the error
```
error: too few arguments to function ‘jl_get_binding_wr’
  109 |     jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var);
```
Replacing line 108 of embedding_wrapper.c by
```
#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 11
    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var, 1);
    jl_checked_assignment(bp, jl_base_module, var, firstarg);
#elif JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 10
```
lets the build finish successfully. This is a trivial combination of the existing code for 1.9 and 1.10.

### PR DESCRIPTION
**Pull request by ufechner7**
Fixes #951 

This PR does the following:
1. Fix `embedding_wrapper.c`
2. Start running CI on Julia 1.11.

Note: for now, we skip the `incremental=false` tests on Julia 1.11 due to #976.

### CODE REVIEW

**Comment by ufechner7** [comment]
The tests for macOS still fail for reasons that I do not understand. Could some have a look at the error messages who knows macOS better than me?

**Comment by PatrickHaecker** [comment]
Thanks for doing the PR!

I am a bit puzzled: To my understanding the code change in embedding_wrapper.c should not have any influence on Julia <= 1.10.
Has it really worked before your PR or is the CI failing on macOS a different issue independent of the 1.11 change?

**Comment by ufechner7** [comment]
CI fails since May 2023, since this commit: https://github.com/JuliaLang/PackageCompiler.jl/commit/33428b2d9c9daf71bbc478a18385358e1b61a137

**Comment by PatrickHaecker** [comment]
Yes, that would be nice. But I suggest that we track that in a different topic.

```
`$(get_julia_cmd()) --sysimage=$sysimage -e 'using Pkg; Pkg.precompile()'`
```
probably has a non-zero return value, but without access to a macOS Julia installation that is indeed hard to impossible to debug.

**Comment by ufechner7** [comment]
I created a separate issue for the CI failures on Mac: https://github.com/JuliaLang/PackageCompiler.jl/actions/runs/10157131523/job/28086727940

**Comment by DilumAluthge** [comment]
@IanButterworth It looks like you modified this file most recently. Would you be able to review this PR?

File: src/embedding_wrapper.c [hunk 1]
```diff
@@ -105,7 +105,10 @@ int main(int argc, char *argv[]) {
     jl_value_t *firstarg = checked_eval_string("popfirst!(ARGS)");
     JL_GC_PUSH1(&firstarg);
     jl_sym_t *var = jl_symbol("PROGRAM_FILE");
-#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 10
+#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 11
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var, 1);
```
Comment (topolarity) [inline src/embedding_wrapper.c:109]:
```suggestion
    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var, /* alloc */ 1);
```

**Review by topolarity — APPROVED** [review]
I can't help bring up PackageCompiler on 1.11 in general, but I can sign off on this change @DilumAluthge

**Comment by DilumAluthge** [comment]
Thanks @topolarity!

**Comment by DilumAluthge** [comment]
The Julia 1.11 CI seems to be failing on all platforms. For example, look at [this CI log](https://github.com/JuliaLang/PackageCompiler.jl/actions/runs/11357992159/job/31595739170?pr=954), which is Julia 1.11, 64-bit, on Linux. (I've removed the verbose listing of all the environment variables in the active process.)

```
error during bootstrap:
LoadError("/tmp/jl_3Qidyp/sysimage_packagecompiler_bfb1e672-8b81-11ef-3d17-c7f4e932fd41.jl", 57, LoadError("/opt/hostedtoolcache/julia/1.11.0/x64/share/julia/stdlib/v1.11/Pkg/src/Pkg.jl", 3, LoadError("/opt/hostedtoolcache/julia/1.11.0/x64/share/julia/stdlib/v1.11/Pkg/src/precompile.jl", 101, Base.IOError("could not spawn setenv(`'' a registries/Registry.tar.gz -tgzip registries/Registry.tar`,[ ... env vars ...]): permission denied (EACCES)", -13))))
_spawn_primitive at ./process.jl:140
_spawn at ./process.jl:157 [inlined]
```

@staticfloat To me, it looks like we are hitting this line in Pkg.jl's `precompile.jl` file:

https://github.com/JuliaLang/Pkg.jl/blob/27c1b1ee5cf15571eb5e54707e812d646ac1dde3/src/precompile.jl#L86-L86

```
cmd = `$(Pkg.PlatformEngines.exe7z()) a "registries/Registry.tar.gz" -tgzip "registries/Registry.tar"`
```

But it seems like `Pkg.PlatformEngines.exe7z()` is returning an empty string?

**Comment by DilumAluthge** [comment]
This `could not spawn` error is appearing in the logs of every failed 1.11 CI job.

**Comment by DilumAluthge** [comment]
I am able to reproduce this error locally on macOS (Apple Silicon):

> ⣠ [00m:57s] PackageCompiler: compiling fresh sysimage (incremental=false)
> "/var/folders/lp/2fm0z_wn3wqf__vj6lc1tcgh0000gq/T/jl_C0bGtL/sysimage_packagecompiler_2694e398-8b8f-11ef-3eef-2781413be4ce.jl", 57, LoadError("/pathtomyjulia/julia/1.11.0/share/julia/stdlib/v1.11/Pkg/src/Pkg.jl", 3, LoadError("/pathtomyjulia/julia/1.11.0/share/julia/stdlib/v1.11/Pkg/src/precompile.jl", 101, Base.IOError("could not spawn setenv(`'' a registries/Registry.tar.gz -tgzip registries/Registry.tar`,[... env vars...]): permission denied (EACCES)", -13))))

**Comment by DilumAluthge** [comment]
Here's an MWE. Obviously, it's not a great MWE (just because it's pretty big), and it would be good to make a smaller MWE, but at least it's a start:

```julia
import PackageCompiler

sysimage_stdlibs = ["Pkg"]

base_sysimage = PackageCompiler.create_fresh_base_sysimage(
    sysimage_stdlibs;
    cpu_target = PackageCompiler.default_app_cpu_target(),
    sysimage_build_args = ``,
)

```

**Comment by DilumAluthge** [comment]
The `base/sysimg.jl` file has changed significantly between 1.10 (https://github.com/JuliaLang/julia/blob/v1.10.5/base/sysimg.jl) and 1.11 (https://github.com/JuliaLang/julia/blob/v1.11.0/base/sysimg.jl). I wonder if that's the source of the problem.

**Comment by DilumAluthge** [comment]
I've opened a separate issue for the bug that is causing Julia 1.11 CI to fail here: #976

**Comment by DilumAluthge** [comment]
Alright, for now, let's skip the `incremental=false` tests on Julia 1.11.

**Comment by DilumAluthge** [comment]
Okay, that change (skipping the `incremental=false` tests on Julia 1.11) seems to have fixed CI.

Once CI has finished running and is all green, I'll merge this PR and make a new release.

**Pull request merged**